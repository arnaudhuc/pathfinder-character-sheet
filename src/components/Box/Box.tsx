/** @jsx jsx */
import { jsx } from '@emotion/core';
import {
	Card,
	CardContent,
	makeStyles,
	Typography,
	Divider,
} from '@material-ui/core';

interface IBox {
	handleClick: () => void;
	children: React.ReactNode;
}

export const Box: React.FC<IBox> = (props) => {
	const { handleClick, children } = props;

	const useStyles = makeStyles({
		root: {
			minWidth: 275,
			width: '100%',
		},
		title: {
			fontSize: 14,
		},
	});

	const classes = useStyles();

	return (
		<Card onClick={handleClick} className={classes.root}>
			<CardContent>
				<Typography component="h2">Test</Typography>
				<Divider variant="fullWidth" />
				{children}
			</CardContent>
		</Card>
	);
};
