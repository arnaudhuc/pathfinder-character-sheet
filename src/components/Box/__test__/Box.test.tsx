/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';

import { Box } from '../Box';

describe('Box', () => {
	test('should render with default props', () => {
		const component = renderer
			.create(
				<Box handleClick={() => true}>
					<p>Children of the box</p>
				</Box>,
			)
			.toJSON();

		expect(component).toMatchSnapshot();
	});
});
