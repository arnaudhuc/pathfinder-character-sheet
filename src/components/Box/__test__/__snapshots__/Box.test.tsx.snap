// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render with default props 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  onClick={[Function]}
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-body1"
    >
      Test
    </h2>
    <hr
      className="MuiDivider-root"
    />
    <p>
      Children of the box
    </p>
  </div>
</div>
`;
